### ISSUE TITLE
`no-async-fn-without-await` rule does not support async iteration (for-await-of loop)

### ISSUE THREAD

**Issue by futpib** [avajs/eslint-plugin-ava#216]
The following is reported even though the function uses `await`:

```js
test('foobar', async t => { // Function was declared as `async` but doesn't use `await`
	for await (const bar of foo) {
		console.log(bar)
	}
});
```
